### PR TITLE
Feature discount resource

### PR DESCRIPTION
**Pull request by ifnull**
Adds Discount resource support

### CODE REVIEW

File: test/fixtures/discounts.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      "usage_limit": null,
+      "minimum_order_amount": "0.00",
+      "applies_to_id": null,
+      "applies_once": false,
+      "discount_type": "fixed_amount",
+      "applies_to_resource": null,
+      "times_used": 0
+    }
+  ]
+}
```
Comment (kevinhughes27) [inline test/fixtures/discounts.json]: picky but can we get the newlines here please!

File: test/discount_test.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        self.fake('discounts/992807812/disable',
+                  method='POST',
+                  code=200,
+                  body=self.load_fixture('discount_disabled'),
+                  headers={'Content-length': '0',
+                           'Content-type': 'application/json'})
+        discount = shopify.Discount.find(992807812)
+        self.assertEqual("enabled", discount.status)
+        discount.disable()
+        self.assertEqual("disabled", discount.status)
```
Comment (kevinhughes27) [inline test/discount_test.py]: nice tests!

**Comment by kevinhughes27** [comment]
code looks good, red travis though :(

**Comment by ifnull** [comment]
@kevinhughes27 I think I know why. I'll take a look tonight.

**Comment by kevinhughes27** [comment]
thanks!

**Comment by ifnull** [comment]
@kevinhughes27 Should be good to go now.

**Pull request merged**